src/persist-txn/src/operator.rs

```diff
@@ -292,6 +295,26 @@ where
                         }
                     }
                     Event::Progress(progress) => {
+                        // If `until.less_equal(progress)`, it means that all
+                        // subsequent batches will contain only times greater or
+                        // equal to `until`, which means they can be dropped in
+                        // their entirety.
+                        //
+                        // Ideally this check would live in
+                        // `txns_progress_source`, but that turns out to be much
+                        // more invasive (requires replacing lots of `T`s with
+                        // `Antichain<T>`s). Given that we've been thinking
+                        // about reworking the operators, do the easy but more
+                        // wasteful thing for now.
+                        if PartialOrder::less_equal(&until, &progress) {
+                            debug!(
+                                "{} progress {:?} has passed until {:?}",
+                                name,
+                                progress.elements(),
+                                until.elements()
+                            );
+                            return;
+                        }
```

Comment:
Is it right that we don't have to look at `Event::Data` here (to filter out updates at times after the `until` too) because that already happens in `persist_source_core` which provides the input to this operator?

Reply:
Correct! Added a comment calling this out